tests/Aquifer.Common.UnitTests/Utilities/StringSimilarityUtilitiesTests.cs

```diff
@@ -0,0 +1,157 @@
+using Aquifer.Common.Utilities;
+
+namespace Aquifer.Common.UnitTests.Utilities;
+
+public sealed class StringSimilarityUtilitiesTests
+{
+    [Theory]
+    [ClassData(typeof(StringDistanceTestData))]
+    public void GivenTwoStringsGetLevenshteinDistance_ShouldReturnCorrectDistance( StringPair strings, int expectedDistance)
+    {
+        var distance = StringSimilarityUtilities.LevenshteinDistance(strings.First, strings.Second);
+        
+        Assert.Equal(expectedDistance, distance);
+    }
+
+    [Theory]
+    [ClassData(typeof(SubStringSplitTestData))]
+    public void LongStringsShouldBeSplitOnLimit_ShouldReturnCorrectSubstrings(SubStringLimit strings, List<string> expectedSubstrings)
+    {
+        var subStrings = StringSimilarityUtilities.GetSubstrings(strings.First, strings.Limit);
+        
+        Assert.Equal(expectedSubstrings, subStrings);
+    }
+    
+    [Theory]
+    [ClassData(typeof(TotalStingDistanceTestData))]
+    public void SplitStringDistanceShouldEqualStringDistance_ShouldReturnCorrectDistance(
+        List<string> substringsListA, 
+        List<string> substringsListB,
+        int expectedDistance)
+    {
+        var totalDistance = StringSimilarityUtilities.ComputeTotalStringDistance(substringsListA, substringsListB);
+        
+        Assert.Equal(expectedDistance, totalDistance);
+    }
+
+    [Theory]
+    [ClassData(typeof(StringSimilarityTestData))]
+    public void StringSimilarity_ShouldReturnCorrectSimilarityAndExecutionTimeMs(
+        StringPair strings, IComparable<double> expectedSimilarity)
+    {
+        var (similarity, _) = StringSimilarityUtilities.ComputeLevenshteinSimilarity(
+            strings.First, 
+            strings.Second, 
+            strings.Limit);
+        
+        Assert.Equal(expectedSimilarity, similarity);
+    }
+}
+
+public class StringPair
+{
+    public required string First { get; set; }
+    public required string Second { get; set; }
+    public int Limit { get; set; }
+}
+
+public class StringDistanceTestData : TheoryData<StringPair, int>
```

Comment:
Each of these classes are only used by a single test method and thus can be moved to `InlineData` attributes on the test method.